src/rubrix/monitoring/model_monitor.py

```diff
@@ -3,16 +3,16 @@
 from typing import Any
 
 from ._spacy import Language, ner_monitor
-from ._transformers import TextClassificationPipeline, classifier_monitor
+from ._transformers import Pipeline, huggingface_monitor
 
 _LOGGER = logging.getLogger(__name__)
 
 
 def monitor(task_model: Any, dataset: str, sample_rate: float = 0.3):
     if isinstance(task_model, Language):
         return ner_monitor(task_model, dataset=dataset, sample_rate=sample_rate)
-    if isinstance(task_model, TextClassificationPipeline):
-        return classifier_monitor(task_model, dataset=dataset, sample_rate=sample_rate)
+    if isinstance(task_model, Pipeline):
+        return huggingface_monitor(task_model, dataset=dataset, sample_rate=sample_rate)
 
     warnings.warn(
         "The provided task model is not supported by monitoring module. "
```

Comment:
In the case of HF pipelines other than text classification and zero-shot, the warning below is not triggered even if the `task_model` is not monitored in the end.

Reply:
I've made some changes covering unsupported models in a more general way. Please, take  a  look